src/binaryen-c.cpp

```diff
@@ -1616,6 +1616,17 @@ double BinaryenConstGetValueF64(BinaryenExpressionRef expr) {
   assert(expression->is<Const>());
   return static_cast<Const*>(expression)->value.getf64();
 }
+const uint8_t* BinaryenConstGetValueV128(BinaryenExpressionRef expr) {
+  if (tracing) {
+    std::cout << "  BinaryenConstGetValueV128(expressions[" << expressions[expr] << "]);\n";
+  }
+
+  auto* expression = (Expression*)expr;
+  assert(expression->is<Const>());
+  void* buf = malloc(16);
```

Comment:
Instead of allocating in this function, it would be better to have the user pass in a buffer to be filled. That way the JS layer could use a stack allocation and avoid the malloc/free entirely. See how `BinaryenLiteralVec128` is implemented and used for an example.

Reply:
Thanks, missed that there's already code doing it this way. Updated accordingly!